airbyte-ci/connectors/cdk/cdk/java_cdk/tasks.py

```diff
@@ -0,0 +1,104 @@
+import os
+
+import pygit2
+from aircmd.actions.environments import with_gradle
+from aircmd.models.base import PipelineContext
+from aircmd.models.utils import load_settings, make_pass_decorator
+from dagger import Client, Container
+from prefect import task
+from prefect.artifacts import create_link_artifact
+
+from .settings import JavaCDKSettings
+
+pass_pipeline_context = make_pass_decorator(PipelineContext, ensure=True)
+pass_global_settings = make_pass_decorator(JavaCDKSettings, ensure=True)
+
+
+
+@task
+async def build_java_cdk_task(client: Client,settings: JavaCDKSettings, ctx: PipelineContext, scan: bool = False) -> Container:
+
+
+    files_from_host=["./build.gradle",
+                "./gradlew",
+                "./gradle", 
+                "./airbyte-cdk/java/**", 
+                "./publish-repositories.gradle", 
+                "./gradle.properties",
+                "./deps.toml",
+                "./build.gradle",
+                "./settings.gradle",
+                "./.env",
+                "./.env.dev",
+                ]
+    
+    repo = pygit2.Repository('.')
+    repo_root = os.path.dirname(os.path.dirname(repo.path))
+    print(repo_root)
+    mount_dir = client.host().directory(repo_root, include=files_from_host)
+    
+    gradle_command = ["./gradlew", ":airbyte-cdk:java:airbyte-cdk:assemble", "publishToMavenLocal", "--build-cache"]
```

Comment:
The general approach here is to only build the runtime code, then publish it to mavenlocal to make it easy to pass into new containers to do later stuff